### DISCUSSION TITLE
[Question] 不同的顶点使用不同的边要怎么做

### DISCUSSION THREAD

**Discussion by liming1010** [apache/incubator-hugegraph#2394] [Q&A] [answered]
### Problem Type (问题类型)

configs (配置项 / 文档相关)

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

server: 1.0
jdk: 11

### Your Question (问题描述)

不同的顶点使用相同的边，比如 用图表示多个家庭：每个家庭有自己的父亲和自己的仔仔，他们有相同的边就是: 父子。创建相同的边的时候就会提示：The edge label '父子' has existed
```
String edge = "父子";
        schema.edgeLabel(edge).sourceLabel("张三")
                .targetLabel("张小明")
                .properties(edge)
                .ifNotExist()
                .create();

        schema.edgeLabel(edge).sourceLabel("李四")
                .targetLabel("李二狗")
                .properties(edge)
                .ifNotExist()
                .create();
```
创建第二个的时候就会出错，不知道描述清楚没有

**Comment by imbajin** [comment, marked as answer]
几个点:
1. 例子里不应该使用人的具体实现作为 vertexLabel, 它们应该是一个 vertex 示例 (label 应该是"Person" 这样的抽象), 虽然只是举例, 也提醒一下
2. 你说的类似的 VertexLabel 用相似的 EdgeLabel, 目前比较简单的做法是, 把 EdgeLabel 改个名字, 例如 `parent" 这样的 edgelabel, 替换为
    - personParentDog
    - dogParentCat
    - `aParentB` 的 edgeLabel 命名方式 (a 和 b 代表两个 vertexLable, 这样可以唯一区分, 语义也清楚)
3. 更好的设计是增加"父子边"的结构设计, 这个 feature 在下一个版本里会发布
<img width="943" alt="image" src="https://github.com/apache/incubator-hugegraph/assets/17706099/b9c44163-99c9-4208-85a8-b423b941563b">

**Comment by liming1010** [comment]
谢谢
